include/nbl/builtin/hlsl/cpp_compat/intrinsics.hlsl

```diff
@@ -191,28 +201,52 @@ inline matrix<T, N, M> inverse(NBL_CONST_REF_ARG(matrix<T, N, M>) m)
 #endif
 }
 
+namespace cpp_compat_intrinsics_impl
+{
+
+	// TODO: concept requiring T to be a float
 template<typename T, typename U>
-inline T lerp(NBL_CONST_REF_ARG(T) x, NBL_CONST_REF_ARG(T) y, NBL_CONST_REF_ARG(U) a)
+struct lerp_helper
 {
+	static inline T lerp(NBL_CONST_REF_ARG(T) x, NBL_CONST_REF_ARG(T) y, NBL_CONST_REF_ARG(U) a)
```

Comment:
I'd constrain that either `is_same_v<T,U> || is_same_v<scalar_type_t<T>,U>` or something similar, you probably don't want a weird `U`  on this spec